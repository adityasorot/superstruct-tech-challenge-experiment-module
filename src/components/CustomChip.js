import { Chip } from "@mui/material";
import React from "react";

const CustomChip = (props) => {
  return (
    <Chip
      sx={{
        color: props.activeColor ? "#03a53c" : "#757575",
        my: { xs: 1, sm: 2 },
        mx: 1,
        borderColor: props.activeColor ? "#03a53c" : "#757575",
        borderRadius: 2,
      }}
      {...props}
    />
  );
};

export default CustomChip;
