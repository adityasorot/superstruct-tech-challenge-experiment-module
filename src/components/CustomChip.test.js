import renderer from "react-test-renderer";
import CustomChip from "./CustomChip";

it("renders active color true", () => {
  const component = renderer.create(
    <CustomChip
      activeColor={true}
      label="VERY VERY VERY LONG (UP TO 35 CHAR)"
      variant="outlined"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
it("renders active color false", () => {
  const component = renderer.create(
    <CustomChip
      activeColor={false}
      label="VERY VERY VERY LONG (UP TO 35 CHAR)"
      variant="outlined"
    />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
