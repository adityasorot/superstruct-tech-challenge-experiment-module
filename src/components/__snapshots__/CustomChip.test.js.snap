// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders active color false 1`] = `
<div
  activeColor={false}
  className="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-1jgvcc3-MuiChip-root"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
  >
    VERY VERY VERY LONG (UP TO 35 CHAR)
  </span>
</div>
`;

exports[`renders active color true 1`] = `
<div
  activeColor={true}
  className="MuiChip-root MuiChip-outlined MuiChip-sizeMedium MuiChip-colorDefault MuiChip-outlinedDefault css-i2nbn3-MuiChip-root"
  onKeyDown={[Function]}
  onKeyUp={[Function]}
>
  <span
    className="MuiChip-label MuiChip-labelMedium css-6od3lo-MuiChip-label"
  >
    VERY VERY VERY LONG (UP TO 35 CHAR)
  </span>
</div>
`;
